newIDE/app/src/MainFrame/Preferences/PreferencesDialog.js

```diff
@@ -242,6 +243,16 @@ const PreferencesDialog = ({ i18n, onClose }: Props) => {
               label={<Trans>Use the new action/condition editor</Trans>}
             />
           </Line>
+          <Line>
+            <Toggle
+              onToggle={(e, check) =>
+                setUseUndefinedVariablesInAutocompletion(check)
+              }
+              toggled={values.useUndefinedVariablesInAutocompletion}
+              labelPosition="right"
+              label={<Trans>Show undefined variables in autocompletion</Trans>}
```

Comment:
I would go for: `Suggest names of variables used in events but not declared in the list of variables`